Microsoft.Azure.Cosmos/src/Resource/Settings/IndexingPolicy.cs

```diff
@@ -111,12 +111,45 @@ public IndexingPolicy()
         [JsonProperty(PropertyName = Constants.Properties.CompositeIndexes)]
         public Collection<Collection<CompositePath>> CompositeIndexes { get; internal set; } = new Collection<Collection<CompositePath>>();
 
+        /// <summary>
+        /// Gets the vector indexes for additional indexes
+        /// </summary>
+        /// <example>
+        /// <![CDATA[
+        /// "vectorIndexes": [
+        ///  {
+        ///    "path": "/vector1",
+        ///    "type": "diskANN"
+        ///  },
+        ///  {
+        ///    "path": "/vector2",
+        ///    "type": "flat "
+        ///  },
+        ///  {
+        ///    "path": "/vector3",
+        ///    "type": "quantizedFlat"
+        ///  }
+        /// ]
+        /// ]]>
+        /// </example>
+        [JsonProperty(PropertyName = "vectorIndexes", NullValueHandling = NullValueHandling.Ignore)]
+        internal Collection<VectorIndexPath> VectorIndexes
+        {
+            get => this.VectorIndexesInternal ??= new Collection<VectorIndexPath>();
+            set => this.VectorIndexesInternal = value;
```

Comment:
```suggestion
            internal set => this.VectorIndexesInternal = value;
```

all the other public collections have internal setters, we don't want the user to do

```
policy.VectorIndexes = new Collection(...)
```

we want them to

```
policy.VectorIndexes.Add(...)
```

Reply:
It is applicable when our contract is `public`:

![image](https://github.com/Azure/azure-cosmos-dotnet-v3/assets/87335885/6b3bd78d-2059-45c0-8cc0-858a7c7e030d)

So, when we finally get to a point to modify this field as `public` I will track this to update the setter to `internal`